### PR TITLE
Feature/sqlalchemy models

### PR DESCRIPTION
**Pull request by CloudNiner**

### CODE REVIEW

**Comment by CloudNiner** [comment]
cc'ing @moradology @kshepard

File: app/ashlar/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+### Migrations
+
+Before creating a migration, read this:
+http://alembic.readthedocs.org/en/latest/autogenerate.html#what-does-autogenerate-detect-and-what-does-it-not-detect
```
Comment (flibbertigibbet) [inline app/ashlar/README.md:4]: I guess they weren't kidding when they say it's "lightweight".

File: app/ashlar/ashlar/models.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    __tablename__ = 'records'
+
+    id = Column(Integer, primary_key=True)
+    label = Column(String(50), nullable=False)
+    slug = Column(String(50), nullable=False)
+    geo = Column(Geometry(geometry_type='POINT', srid=4326))
+    schema_id = Column(Integer, ForeignKey('record_schemas.id'))
+    json_data = Column(JSONB)
+
+    __table_args__ = (Index('json_data_idx', 'json_data', postgresql_using='gin'),)
```
Comment (flibbertigibbet) [inline app/ashlar/ashlar/models.py:42]: Trailing commas in this file intentional?

File: app/ashlar/migrations/env.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+fileConfig(config.config_file_name)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+from ashlar import models
+target_metadata = models.BaseModel.metadata
+# target_metadata = None
```
Comment (flibbertigibbet) [inline app/ashlar/migrations/env.py]: Remove commented `target_metatdata` here?

File: app/ashlar/ashlar/models.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+BaseModel = declarative_base()
+
+
+class AshlarModel(BaseModel):
+    __abstract__ = True
+
+    created = Column(DateTime(timezone=True), nullable=False, default=func.now())
+    modified = Column(DateTime(timezone=True), nullable=False, server_onupdate=func.now(), default=func.now())
+    occurred_from = Column(DateTime(timezone=True), nullable=False, default=func.now())
+    occurred_to = Column(DateTime(timezone=True), nullable=False, default=func.now())
```
Comment (ddohler) [inline app/ashlar/ashlar/models.py]: Consider moving these onto `Record`; I'm not sure it makes sense for them to be inherited by `RecordSchema` and `ItemSchema`.

**Comment by flibbertigibbet** [comment]
+1 from me, not knowing about app model/SQLAlchemy usage

File: app/ashlar/ashlar/models.py (hunk 2 above)
Comment (CloudNiner) [inline app/ashlar/ashlar/models.py:42]: Yeah, so this gets interpreted as tuple of tuples.

File: app/ashlar/README.md (hunk 1 above)
Comment (CloudNiner) [inline app/ashlar/README.md:4]: Ha yup. I had to modify this first migration to not attempt to delete the postgis spatial_ref_sys tables.

File: app/ashlar/ashlar/models.py (hunk 4 above)
Comment (CloudNiner) [inline app/ashlar/ashlar/models.py]: Yeah, good point. Assuming you're referring to the `occurred_` fields. The created/modified fields are valuable for all models.

Comment (ddohler) [inline app/ashlar/ashlar/models.py]: Yeah, sorry, I put that comment in an unclear spot. I did mean just the `occurred_*` fields.

**Comment by CloudNiner** [comment]
Updated with better model definitions. Might want to take another look before merging.

@flibbertigibbet @ddohler

**Comment by flibbertigibbet** [comment]
I'm still good with this.

**Comment by ddohler** [comment]
Me too.

**Comment by kshepard** [comment]
LGTM

**Comment by moradology** [comment]
+1 here as well, we can revisit which type of index to use later but 'gin' is the most feature-rich, so I think it's the best bet.

**Pull request merged**